activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -21,8 +21,9 @@ def query(sql, name = nil) # :nodoc:
           end
         end
 
+        copy_to = /copy\s+(.*)\s+to\s+(.*)/i
```

Comment:
:thinking: It is still possible to bypass this using for example `COPY ex (id) /* TO */ FROM STDIN;` or `COPY ex ( to ) FROM STDIN;`. Parsing SQL via regexp is probably not good idea in general.

Reply:
You are probably right, however, what you are suggesting seems to be a much bigger change that might need discussed a bit more. 

What do you think of this?

```
copy\s+.*\(.*\)\s+to\s+.*
```

Turns out I forgot to escape the `( )` which will be required regardless if you use a tablename and columns or a query. Without these it will throw a syntax error.

Reply:
That would not support (for example) column-less format aka `COPY ex TO STDOUT;` if I understand it well. I still think regexp parsing SQL is not the way in here. :thinking:

Reply:
I really appreciate all the use cases you are thinking of. I also apologize if I seem to just be iterating on the regex, which I think is the least amount of work we can do to support this type of query for postgres 😬 and I am in no way dismissing what you are saying about not using regex.